### ISSUE TITLE
Wallaby resulting with failing tests that are not failures actually.

### ISSUE THREAD

**Issue by bibonev** [wallabyjs/public#756]
### Issue description or question

Hello, 

I have a very strange problem with running wallaby in VS Code. Wallaby says that I have number of failures with the tests and when I jump to the failing ones, I just have to add a new line or make some minor change, save the file and the failure is gone. What can cause this?
### Wallaby.js configuration file

``` javascript
module.exports = (wallaby) => {
  return {
    files: [
      './test/.setup.js',
      'src/**/*.js',
      'src/**/*.json'
    ],

    tests: [
      'test/**/*spec.js'
    ],

    env: {
      type: 'node',
      runner: 'node' 
    },

    testFramework: 'mocha',

    compilers: {
      '**/*.js': wallaby.compilers.babel({
        babelrc: false,
        presets: ['es2015', 'react', 'stage-2'],
        plugins: ['syntax-async-generators', 'add-module-exports', 'transform-class-properties']
      })
    },

    setup: function() {
          require('./test/.setup.js');
    }
  };
};
```
### Code editor or IDE name and version

Visual Studio Code v1.4.0
### OS name and version

Windows

**Comment by ArtemGovorov** [comment]
We have seen issues like this before and in all cases the reason was a dependency between tests. Unlike mocha, that runs all tests, wallaby runs tests in parallel, so neither the execution order, nor the test files that will be executed together can't be guaranteed. 

From what you've described, some of your tests may fail because they somehow depend on other tests. When you edit a specific files, wallaby runs just that file, so nothing influences its tests and it passes. 

If you don't see any issues with your test dependencies, please create a sample where the issue can be reproduced, I'm happy to have a look.

**Comment by ArtemGovorov** [comment]
Is the issue easily reproducible? What errors are you getting in those specs that first fail and then pass after small modifications?

I'd start with setting [both `workers` count to one](https://wallabyjs.com/docs/config/workers.html) to see if the issue is still there and remove the parallelisation factor before it's sorted out. 

Then I'd try to modify `tests` setting to only include the spec where the fantom error shows up. Running the config with just that spec should clear the error. Then I'd go ahead and include some other specs, while restarting wallaby to see if the issue shows up again (like bisecting by the other way around). The idea is basically to find out what other spec is influencing this one.

**Comment by ivankolev** [comment]
This happens to me as well. I have set the workers values to 1 which didn't change the result, it still fails under wallaby while regular Karma run configuration in IDEA passes all. Notably, the karma run uses very old version of node, 0.12.7 and a different version of phantom js (2.1.1) than the packaged in wallaby, 1.9.8, so that might be a cause...

**Comment by ArtemGovorov** [comment]
@ivankolev You may try using [the same PhantomJs version in Wallaby as in Karma](https://wallabyjs.com/docs/integration/phantomjs2.html), or even [Electron test runner](https://wallabyjs.com/docs/integration/electron.html). But I doubt that it's the issue, looks more like a dependency between tests.

Is the issue easily reproducible? What errors are you getting in those specs that first fail and then pass after small modifications? Did you try only specifying one spec (with the issue) in the `tests` list?

**Comment by ivankolev** [comment]
@ArtemGovorov you are right, changing the phantomjs did not solve it. It still fails even if I do load only this spec. Something about $httpBackend, but I can't see how it runs fine in Karma and not in wallaby...
Anyways, thanks for the pointers

**Comment by ArtemGovorov** [comment]
@ivankolev 
> Something about $httpBackend, but I can't see how it runs fine in Karma and not in wallaby

Most likely it's just a small difference in the configs between those two. Given that you've isolated the spec, and the error is reproducible, perhaps you could create and share a sample project with just that spec and karma/wallaby configs to run it? With the sample, I'm pretty confident I'll resolve the issue quickly for you whatever it is.

**Comment by ivankolev** [comment]
I'll try to create a sample but that might prove problematic. In any case, much appreciated the support

**Comment by IPWright83** [comment]
I've seen issues like this before as-well. I've even seen the IDE and the wallabyjs.com/app reporting different results which is weird. @ArtemGovorov I'd be interested to know, what gets shared between tests exactly? (I'm wondering about my tests that check events on the DOM and whether they might interact).

**Comment by ArtemGovorov** [comment]
> I'd be interested to know, what gets shared between tests exactly?

Unfortunately it's not possible to tell without seeing a sample where I could reproduce it.

> IDE and the wallabyjs.com/app reporting different results

It may be a different type of issue (like a bug in wallaby app). A sample and some steps to reproduce are greatly appreciated too.

**Comment by IPWright83** [comment]
@ArtemGovorov I've struggled to get a reproduction unfortunately. If I do, I'll send you one over, but normally I just manually trigger a re-run of all.

Regarding the sharing, I meant is each test run in it's own sandbox? do they share the same `window` object or even `document` object?

**Comment by ArtemGovorov** [comment]
@IPWright83 

>Regarding the sharing, I meant is each test run in it's own sandbox? do they share the same window object or even document object?

Oh, sorry, I didn't initially understood your question. Yes, in most of cases, tests share window and document object. They basically share one HTML sandbox page they are running in. Just like in karma or plain jasmine/mocha with an HTML page. "In most of cases" because sometimes wallaby.js runs tests in parallel, so the test files are grouped into a few chunks are run in a few sandboxes, each group in a complete isolation from others.

**Comment by IPWright83** [comment]
@ArtemGovorov I'm not sure if you're about, but at the moment I've got my wallabyjs.com/app/#/tests out of sync with my IDE ever after restarting wallaby in the IDE. Unfortunately I have no reliable reproduction (e.g. if I restart my machine, it'll probably work again fine for a while). I don't know if there's any info I can gather while in this state to help you fix that? Also do you want me to log another issue for it?

**Comment by ArtemGovorov** [comment]
Could you please make a screenshot to highlight the difference, open Chrome Dev tools to see if there're any errors for the Wallaby App page, and then try refreshing the the wallaby app page to see if it syncs or not?

**Comment by IPWright83** [comment]
@ArtemGovorov sure - do you use Skype or anything? Wondering if direct comms might be easier?

**Comment by IPWright83** [comment]
@ArtemGovorov note that I've created a new issue - conscious of hijacking this thread https://github.com/wallabyjs/public/issues/798

**Comment by ArtemGovorov** [comment]
Closing the issue for now, please let me know if it's still valid and you can share a sample project so I may have a look.
